lib/oper1.g

```diff
@@ -601,6 +601,27 @@ BIND_GLOBAL( "INSTALL_METHOD",
       fi;
     fi;
 
+    if IS_FUNCTION(method) and not HAS_NAME_FUNC(method) then
+        funcname := SHALLOW_COPY_OBJ(NAME_FUNC(opr));
+        APPEND_LIST_INTR(funcname, " ");
+        if info <> false then
+            APPEND_LIST_INTR(funcname, info);
+        else
+            APPEND_LIST_INTR(funcname, "method");
+        fi;
+        SET_NAME_FUNC(method, funcname);
+    fi;
+
+    if IS_FUNCTION(rank) and not HAS_NAME_FUNC(rank) then
+        funcname := "Priority calculation for ";
+        APPEND_LIST_INTR(funcname, NAME_FUNC(opr));
+        if info <> false then
+            APPEND_LIST_INTR(funcname, " ");
+            APPEND_LIST_INTR(funcname, info);
+        fi;
+        SET_NAME_FUNC(rank, funcname);
+    fi;
```

Comment:
Aha, I overlooked that you added this code. I'll have to read and understand that now, then.

Reply:
OK, so this makes sense, I guess -- but there really should be tests that demonstrate this new feature, and also ensure it works and keeps working.

Also, PR #3430 is ready, and I kinda prefer that over the hackish addition of `HasNameFunction`, but of course I am biased. Perhaps we can get some other people to chime in with reviews.